BotRoyaleUnity/Assets/Scripts/RoombaMovement.cs

```diff
@@ -71,29 +78,20 @@ void FixedUpdate()
             speed = rigidBody.velocity.magnitude;
             if (speed < 10)
             {
-                if (navigationMode != "reverse")
-                {
-                    strongest = gameObject.GetComponent<PartHandler>().greatestDirectionStrength();
+                float arenaScale = arena.GetComponent<ShrinkArena>().GetLocalScale().x;
+                strongest = gameObject.GetComponent<PartHandler>().greatestDirectionStrength();
+                if(navigationMode == "target"){
                     Vector3 direction = Quaternion.AngleAxis(strongest * -90 - 90, Vector3.up) * transform.forward;
                     rigidBody.AddForce(direction * (rigidBody.mass * 15));
-                    if (speed > 1)
-                    {
-                        stuckTimer = Time.time;
-                    }
-                    else if (Time.time - stuckTimer > 1)
-                    {
-                        SetNavigationMode("reverse");
-                        stuckTimer = Time.time;
-                    }
-                }
-                else
-                {
-                    rigidBody.AddForce(-transform.forward * 10 * rigidBody.mass);
-                    if (Time.time - stuckTimer > 0.55)
-                    {
-                        SetNavigationMode("target");
-                        stuckTimer = Time.time;
-                    }
+                }else if(navigationMode == "run"){
+                    float distanceFromClosest = (attack.transform.position - transform.position).magnitude;
```

Comment:
This throws errors while the game is running. `NullReferenceException: Object reference not set to an instance of an object
RoombaMovement.FixedUpdate () (at Assets/Scripts/RoombaMovement.cs:87)`